packages/dash.rb

```diff
@@ -2,12 +2,12 @@
 
 class Dash < Package
   description 'The Debian Almquist Shell (dash) is a POSIX-compliant shell derived from ash that executes scripts faster than bash and has fewer library dependencies.'
-  homepage 'https://salsa.debian.org/debian/dash'
+  homepage 'https://salsa.debian.org/debian/dash/'
   version '0.5.11.1'
   license 'BSD'
   compatibility 'all'
-  source_url 'https://salsa.debian.org/debian/dash/-/archive/upstream/0.5.11.1/dash-upstream-0.5.11.1.tar.bz2'
-  source_sha256 '4bb75944bb47fa6d1cf1e8a3ff941f6f1fb23497b553446e9f615d52d07ef1e7'
+  source_url 'https://salsa.debian.org/debian/dash.git'
+  git_tag 'upstream/0.5.11.2'
```

Comment:
Can't tags, branches and hashes all use the same variable generically?

Reply:
branches and tags use the same variable, but hashes need their own. One needs to specify the tag that the hash is on.

Reply:
A tag is just a human-readable alias for a hash.  I'm not understanding the logic.

Reply:
Okay. `-b` allows you to specifically clone a branch or tag. `-b` doesn't work on hashes.

`git_tag` works with either a branch or tag.
`git_hash` works with a tag or hash.

It's more bandwidth efficient to use `git_tag` for tags.

Reply:
@saltedcoffii @uberhacker with the aforementioned logic I shared, the same variable can be used for tags, branches, and hashes.

"_logic_":
```
    # Turn off git warnings.
    system 'git config --global advice.detachedHead false'
    system 'git config --global init.defaultBranch main'
    @git_dir = 'git_dir_for_cloning'
    @git_hash = 'some_tag_or_branch_or_hash'
    @git_url = 'https://github.com/chromebrew/crew-profile-base'
    FileUtils.rm_rf(@git_dir)
    FileUtils.mkdir_p(@git_dir)
    Dir.chdir @git_dir do
      system 'git init'
      system "git remote add origin #{@git_url}"
      system "git fetch --depth 1 origin #{@git_hash}"
      system 'git checkout FETCH_HEAD'
```

The `@git_hash` variable here covers hash, tag, or branch, as I understand it. I don't really care what the variable is called.

I don't have a solution for slow internet though... Other than to say, we've all been there. 

But this should be optimized for end user building, and for the end user, there's no reason to download an entire git clone... this should be MUCH faster,